app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/dlc/DLCStatus.scala

```diff
@@ -1,360 +1,676 @@
+          DLCTimeouts(contractMaturity, contractTimeout),
+          feeRate,
+          totalCollateral,
+          localCollateral,
+          fundingTxId,
+          closingTxId,
+          oracleSigs,
+          outcome
+        )
+      case DLCState.RemoteClaimed =>
+        require(oracleSigs.size == 1,
+                "Remote claimed should only have one oracle sig")
+        RemoteClaimed(
+          paramHash,
+          isInitiator,
+          tempContractId,
+          contractId,
+          oracleInfo,
+          ContractInfo.fromTLV(contractInfoTLV),
+          DLCTimeouts(contractMaturity, contractTimeout),
+          feeRate,
+          totalCollateral,
+          localCollateral,
+          fundingTxId,
+          closingTxId,
+          oracleSigs.head,
+          outcome
+        )
+      case DLCState.Refunded =>
+        Refunded(
+          paramHash,
+          isInitiator,
+          tempContractId,
+          contractId,
+          oracleInfo,
+          ContractInfo.fromTLV(contractInfoTLV),
+          DLCTimeouts(contractMaturity, contractTimeout),
+          feeRate,
+          totalCollateral,
+          localCollateral,
+          fundingTxId,
+          closingTxId
+        )
+
     }
   }
 
-  /** The state where the remote party has successfully
-    * executed a unilateral execution by broadcasting both
-    * a CET and successfully spending their to_local output.
-    */
-  case class RemoteClaimed(
-      eventId: Sha256DigestBE,
+  def calculateOutcomeAndSig(
       isInitiator: Boolean,
       offer: DLCOffer,
       accept: DLCAccept,
       sign: DLCSign,
-      fundingTx: Transaction,
-      cet: Transaction,
-      closingTx: Transaction)
-      extends AcceptedDLCStatus {
-    override val statusString: String = "REMOTE CLAIMED"
-  }
+      cet: Transaction): (SchnorrDigitalSignature, DLCOutcomeType) = {
+    val cetSigs = cet
+      .asInstanceOf[WitnessTransaction]
+      .witness
+      .head
+      .asInstanceOf[P2WSHWitnessV0]
+      .signatures
+
+    require(cetSigs.size == 2,
+            s"There must be only 2 signatures, got ${cetSigs.size}")
+
+    val oraclePubKey = offer.oracleInfo.pubKey
+    val rVals = offer.oracleInfo.nonces
+
+    def aggregateR(numSigs: Int): SchnorrNonce = {
+      rVals.take(numSigs).map(_.publicKey).reduce(_.add(_)).schnorrNonce
+    }
 
-  /** The state where the remote party has published a
-    * CET but failed to spend their to_local output
-    * and this party has successfully spent that to_local
-    * output with a penalty transaction.
-    */
-  case class RemotePenalized(
-      eventId: Sha256DigestBE,
-      isInitiator: Boolean,
-      offer: DLCOffer,
-      accept: DLCAccept,
-      sign: DLCSign,
-      fundingTx: Transaction,
-      cet: Transaction,
-      penaltyTx: Transaction)
-      extends AcceptedDLCStatus {
-    override val statusString: String = "REMOTE PENALIZED"
-  }
+    def sigFromMsgAndSigs(
```

Comment:
So here is my best attempt at writing a scaladoc for this

```scala
    /**
     * Completes the given adaptor signature for the given outcome
     * @param outcome the outcome we are signing
     * @param adaptorSig the adaptor signature given from the counterparty that we need to complete
     * @param cetSig our local CET sig that we use
     * @return
     */
    def sigFromMsgAndSigs(
        outcome: DLCOutcomeType,
        adaptorSig: ECAdaptorSignature,
        cetSig: ECDigitalSignature): SchnorrDigitalSignature
```

i'm pretty sure that is wrong though. Could you correct my misconceptions and write an accurate scaladoc? Maybe using the framework i laid out in mine.

Reply:
Could we also move this method `sigFromMsgAndSigs` and `aggregateR` out of `calculateOutcomeAndSig` and into private helper methods?

Reply:
Added moving methods to todo list